ext/gd/gd.c

```diff
@@ -3135,19 +3135,18 @@ static void php_imagettftext_common(INTERNAL_FUNCTION_PARAMETERS, int mode, int
 	double ptsize, angle;
 	char *str = NULL, *fontname = NULL;
 	char *error = NULL;
-	int argc = ZEND_NUM_ARGS();
 	gdFTStringExtra strex = {0};
 
 	if (mode == TTFTEXT_BBOX) {
-		if (argc < 4 || argc > ((extended) ? 5 : 4)) {
-			ZEND_WRONG_PARAM_COUNT();
-		} else if (zend_parse_parameters(argc, "ddss|a", &ptsize, &angle, &fontname, &fontname_len, &str, &str_len, &EXT) == FAILURE) {
+		if (extended && zend_parse_parameters(ZEND_NUM_ARGS(), "ddss|a!", &ptsize, &angle, &fontname, &fontname_len, &str, &str_len, &EXT) == FAILURE) {
+			RETURN_THROWS();
+		} else if (zend_parse_parameters(ZEND_NUM_ARGS(), "ddss", &ptsize, &angle, &fontname, &fontname_len, &str, &str_len) == FAILURE) {
```

Comment:
I think we should instead make `imageftbbox` and `imagettfbbox` aliases and accept the `EXT` parameter on both.

Reply:
Also, can we make the default value empty array rather than accepting null?

Reply:
@cmb69 Are you ok with these ideas?

> Also, can we make the default value empty array rather than accepting null?

Yeah, I wanted to ask the same. There seems to be no difference between `null` as `[]`.

Reply:
In PHP 7, `null` wasn't supported at all, so defaulting to `[]` makes sense.